core/objlist_data.hpp

```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <vector>
+
+namespace husky {
+template <typename ObjT> class ObjListData {
+    template <typename T> friend class ObjList;
```

Comment:
@ClydeZhao Yes, you are right. It is not inheritance relationship. And I also don't recommend using private inheritance. 

@ddmbr Only two solutions:
1. Make all the data members public; 
2. Use friend